spicy/toolchain/src/compiler/codegen/parser-builder.cc

```diff
@@ -1828,3 +1840,27 @@ void ParserBuilder::finishLoopBody(const Expression& cookie, const Location& l)
     pushBuilder(std::move(body),
                 [&]() { parseError("loop body did not change input position, possible infinite loop", l); });
 }
+
+void ParserBuilder::feature(const type::Unit& unit, std::string_view feature, std::function<void()> f) {
+    // FIXME(bbannier): Only handling for the `is_filter` and
+    // `supports_filters` feature flags is currently implemented.
+    if ( feature != "is_filter" && feature != "supports_filters" ) {
+        f();
+        return;
+    }
+
```

Comment:
Let's also annotate users of other feature-dependent code so we can emit conditional code for all of them.

Reply:
I simplified this PR to only work on `%random-access` features, and will handle the other features in a follow-up PR.